### ISSUE TITLE
ESP8266 compiler errors

### ISSUE THREAD

**Issue by kleinmantara** [MajicDesigns/MD_YX5300#1]
Nice work - with the Arduino IDE and a UNO it works. But with a esp8266 i got the following compile errors:
```^
lib/MD_YX5300/src/MD_YX5300.cpp: In member function 'bool MD_YX5300::sendRqst(MD_YX5300::cmdSet_t, uint8_t, uint8_t)':
lib/MD_YX5300/src/MD_YX5300.cpp:133:6: error: request for member 'write' in '8', which is of non-class type 'int'
_S.write(msg, ARRAY_SIZE(msg));
^
lib/MD_YX5300/src/MD_YX5300.cpp:138:40: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
dumpMessage(msg, ARRAY_SIZE(msg), "S");

^
lib/MD_YX5300/src/MD_YX5300.cpp: At global scope:
lib/MD_YX5300/src/MD_YX5300.cpp:153:54: error: default argument given for parameter 1 of 'void MD_YX5300::processResponse(bool)' [-fpermissive]
void MD_YX5300::processResponse(bool bTimeout = false)
^
In file included from lib/MD_YX5300/src/MD_YX5300.cpp:24:0:
lib/MD_YX5300/src/MD_YX5300.h:942:8: error: after previous specification in 'void MD_YX5300::processResponse(bool)' [-fpermissive]
void processResponse(bool bTimeout = false);                ///< Process a serial response message
^
lib/MD_YX5300/src/MD_YX5300.cpp: In member function 'void MD_YX5300::processResponse(bool)':
lib/MD_YX5300/src/MD_YX5300.cpp:156:35: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
dumpMessage(_bufRx, _bufIdx, "R");
^
lib/MD_YX5300/src/MD_YX5300.cpp:166:26: error: invalid conversion from 'uint8_t {aka unsigned char}' to 'MD_YX5300::status_t' [-fpermissive]
_status.code = _bufRx[3];
^
*** [.pioenvs/esp12e/lib55b/MD_YX5300/MD_YX5300.cpp.o] Error 1
```

**Comment by MajicDesigns** [comment]
Duplicate _S in the ESP8266 library. Fixed v1.0.1

**Comment by thijstriemstra** [comment]
is ESP32 also supposed to work? Opened #8

**Comment by kleinmantara** [comment]
Hi, in the current master the "_S" bug is there again on a esp8266.

**Comment by MajicDesigns** [comment]
Fixed (again) and released in v1.3.1

Interesting that the clash is now with a variable that is private in the class. ESP8266 must have this defined as a macro.
